api-participant/src/main/java/bio/terra/pearl/api/participant/service/ParticipantTaskExtService.java

```diff
@@ -1,67 +1,94 @@
-  public List<TaskAndSurvey> listSurveyTasks(
-      ParticipantUser user, String portalShortcode, EnvironmentName envName, TaskType taskType) {
+  public TasksAndForms listAllTasksAndForms(
+      ParticipantUser user, String portalShortcode, EnvironmentName envName) {
     PortalWithPortalUser portalUser =
         authUtilService.authParticipantToPortal(user.getId(), portalShortcode, envName);
     List<Enrollee> participantEnrollees =
         enrolleeService.findByPortalParticipantUser(portalUser.ppUser());
 
-    List<ParticipantTask> outreachTasks =
+    Map<TaskType, List<ParticipantTask>> tasksByType =
         participantTaskService
             .findByEnrolleeIds(participantEnrollees.stream().map(Enrollee::getId).toList())
             .values()
             .stream()
             .flatMap(Collection::stream)
-            .filter(task -> taskType == null || task.getTaskType().equals(taskType))
-            .toList();
+            .collect(Collectors.groupingBy(ParticipantTask::getTaskType));
 
-    List<Survey> outreachSurveys =
-        surveyService.findByStableIds(
-            outreachTasks.stream().map(ParticipantTask::getTargetStableId).toList(),
-            outreachTasks.stream().map(ParticipantTask::getTargetAssignedVersion).toList());
-    List<TaskAndSurvey> tasksAndSurveys =
-        IntStream.range(0, outreachTasks.size())
-            .mapToObj(i -> new TaskAndSurvey(outreachSurveys.get(i), outreachTasks.get(i)))
-            .toList();
-    return tasksAndSurveys;
+    List<TaskAndForm> researchTaskAndSurveys =
+        getTasksAndFormsByType(TaskType.SURVEY, surveyService::findByStableIds, tasksByType);
+
+    List<TaskAndForm> outreachTaskAndSurveys =
+        getTasksAndFormsByType(TaskType.OUTREACH, surveyService::findByStableIds, tasksByType);
+
+    List<TaskAndForm> consentTaskAndSurveys =
+        getTasksAndFormsByType(TaskType.CONSENT, consentFormService::findByStableIds, tasksByType);
+
+    return new TasksAndForms(researchTaskAndSurveys, outreachTaskAndSurveys, consentTaskAndSurveys);
+  }
+
+  private <F extends VersionedForm> List<TaskAndForm> getTasksAndFormsByType(
+      TaskType taskType,
+      BiFunction<List<String>, List<Integer>, List<F>> findByStableIds,
+      Map<TaskType, List<ParticipantTask>> tasksByType) {
+
+    List<ParticipantTask> tasks = tasksByType.getOrDefault(taskType, List.of());
+    List<F> forms =
+        findByStableIds.apply(
+            tasks.stream().map(ParticipantTask::getTargetStableId).toList(),
+            tasks.stream().map(ParticipantTask::getTargetAssignedVersion).toList());
+
+    return IntStream.range(0, tasks.size())
+        .mapToObj(i -> new TaskAndForm(forms.get(i), tasks.get(i)))
+        .toList();
   }
 
-  public record TaskAndSurvey(Survey survey, ParticipantTask task) {}
+  public record TaskAndForm(VersionedForm form, ParticipantTask task) {}
+
+  public record TasksAndForms(
```

Comment:
I don't think there's any need to have a DTO that breaks these out -- they can be sorted/binned client-side